config/cmake/ucm.cmake

```diff
@@ -32,18 +15,18 @@ cmake_minimum_required(VERSION 2.8.12)
 include(CMakeParseArguments)
 
 # optionally include cotire - the git submodule might not be inited (or the user might have already included it)
-#if(NOT COMMAND cotire)
-#    include(${CMAKE_CURRENT_LIST_DIR}/../cotire/CMake/cotire.cmake OPTIONAL)
-#endif()
-#
-#if(COMMAND cotire AND "1.7.9" VERSION_LESS "${COTIRE_CMAKE_MODULE_VERSION}")
-#    set(ucm_with_cotire 1)
-#else()
-#    set(ucm_with_cotire 0)
-#endif()
-#
-#option(UCM_UNITY_BUILD          "Enable unity build for targets registered with the ucm_add_target() macro"                     OFF)
-#option(UCM_NO_COTIRE_FOLDER     "Do not use a cotire folder in the solution explorer for all unity and cotire related targets"  ON)
+if(NOT COMMAND cotire)
```

Comment:
I think I had commented most of this out since it wasn't in use and was cluttering up the build somewhat

Reply:
we probably want it commented out for the time being since it would be confusing

Reply:
What do you mean by cluttering up the build?

Reply:
we are not using Cotire,  we don't have any plans to use cotire, so searching for that and any options there are pointless,    the UCM_UNIT_BUILD and UCM_NO_COTIRE_FOLDER are then options that are meaningless and should not show up in the cmake-gui or in the cache.  They are not meaningful for the project, so I don't think they should even show up.

Reply:
Having less custom changes for future updates would be nice.

How about something like:
```
if(COMMAND cotire AND "1.7.9" VERSION_LESS "${COTIRE_CMAKE_MODULE_VERSION}")
    set(ucm_with_cotire 1)
    option(UCM_UNITY_BUILD          "Enable unity build for targets registered with the ucm_add_target() macro"                     OFF)
    option(UCM_NO_COTIRE_FOLDER     "Do not use a cotire folder in the solution explorer for all unity and cotire related targets"  ON)
else()
    set(ucm_with_cotire 0)
endif()
```

I think that might be a change upstream would accept.

Reply:
For our purposes I don't really even want to search for COTIRE.  which happens on the earlier line.

Reply:
It checks a fixed location for the existence of a file and doesn't output anything if the file isn't found -- that should take a fraction of a second.

Reply:
ok

Reply:
Another option that might be possible to get upstream would be surrounding the entire cotire block with a check for "UCM_DISABLE_COTIRE" -- which could be set right before including ucm.

Reply:
Any preference? That could have some benefit in a scenario with subprojects using cotire, allowing individual projects to opt-in/out.

Reply:
That would be acceptable I think.  Have you ever used cotire?   Have you ever seen anyone use cotire?

Reply:
I've seen it included in some projects, though cmake 3.16 adds precompiled header support and an upcoming version might add unity build support.

Reply:
It looks like it is an opt-in feature -- `ucm_add_target` needs to get used instead of the built-in CMake `add_library` or `add_executable` functions -- so there probably isn't much value in having an option for (sub)projects completely disabling the check to see if it exists.

I changed it to only add the options if cotire is available, so it won't show up for building HELICS.